src/box/index.cc

```diff
@@ -99,8 +99,7 @@ error_set_index(struct error *error, const struct index_def *index_def)
 {
 	error_set_str(error, "index", index_def->name);
 	error_set_uint(error, "index_id", index_def->iid);
-	struct space *space = space_by_id(index_def->space_id);
-	error_set_str(error, "space", space->def->name);
+	error_set_str(error, "space", index_def->space_def->name);
```

Comment:
There are a few other places where we look up a space just to get its name, for example:

https://github.com/tarantool/tarantool/blob/1f75231a0c0ce062e08660e0bc51485f7086412c/src/box/index.cc#L217-L221

Please find and fix them all.